src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -3302,8 +3316,8 @@ void M()
     {
         var d = new Dictionary<int, int>
         {
-            [1] = 4,
-        };$$
+            [1];$$ = 4,
```

Comment:
hmm... this seems weird to me. Is this a valid semicolon placement? If not, why not just leave the caret where it was?

Reply:
The new location is where I would expect it to appear for now. We don't consider the braces for an object initializer to be enclosing delimiters; once we do, the semicolon will go back to outside them.